osbuild/__init__.py

```diff
@@ -116,13 +123,25 @@ def run(self, argv, binds=None, readonly_binds=None, **kwargs):
             f"--machine={self.machine_name}",
             f"--directory={self.root}",
             f"--bind={libdir}/osbuild-run:/run/osbuild/osbuild-run",
-            *[f"--bind={b}" for b in (binds or [])],
+            f"--bind-ro={libdir}/osbuild:/run/osbuild/osbuild",
+            *[f"--bind-ro={b}" for b in [self.api + ":" + "/run/osbuild/api",
+                                         *(binds or [])]],
             *[f"--bind-ro={b}" for b in [argv[0] + ":" + command,
                                          *(readonly_binds or [])]],
             "/run/osbuild/osbuild-run",
             command
             ] + argv[1:], **kwargs)
 
+    @contextlib.contextmanager
+    def bound_socket(self, name):
+        sock = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+        sock_path = os.path.join(self.api, name)
+        sock.bind(os.path.join(self.api, name))
+        try:
+            yield sock
+        finally:
+            os.unlink(sock_path)
```

Comment:
Should we close the `sock` here or do you expect callers to hold on to it?

Reply:
Good point. Fixed.